client/ayon_hiero/api/pipeline.py

```diff
@@ -35,42 +51,80 @@
+):
+    name = "hiero"
 
-    log.info("Registering Hiero plug-ins..")
-    pyblish.register_host("hiero")
-    pyblish.register_plugin_path(PUBLISH_PATH)
-    register_loader_plugin_path(LOAD_PATH)
-    register_creator_plugin_path(CREATE_PATH)
+    def open_workfile(self, filepath):
+        return open_file(filepath)
 
-    # register callback for switching publishable
-    pyblish.register_callback("instanceToggled", on_pyblish_instance_toggled)
+    def save_workfile(self, filepath=None):
+        return save_file(filepath)
 
-    # install menu
-    menu.menu_install()
-    menu.add_scripts_menu()
+    def work_root(self, session):
+        return work_root(session)
 
-    # register hiero events
-    events.register_hiero_events()
+    def get_current_workfile(self):
+        return current_file()
 
+    def workfile_has_unsaved_changes(self):
+        return has_unsaved_changes()
 
-def uninstall():
-    """
-    Uninstalling Hiero integration for avalon
+    def get_workfile_extensions(self):
+        return file_extensions()
 
-    """
-    log.info("Deregistering Hiero plug-ins..")
-    pyblish.deregister_host("hiero")
-    pyblish.deregister_plugin_path(PUBLISH_PATH)
-    deregister_loader_plugin_path(LOAD_PATH)
-    deregister_creator_plugin_path(CREATE_PATH)
+    def get_containers(self):
+        return ls()
+
+    def install(self):
+        ''' Installing all requarements for hiero host
+        '''
+
+        # adding all events
+        events.register_events()
+
+        log.info("Registering Hiero plug-ins..")
+        pyblish.register_host("hiero")
+        pyblish.register_plugin_path(PUBLISH_PATH)
+        register_loader_plugin_path(LOAD_PATH)
+        register_creator_plugin_path(CREATE_PATH)
+
+        # register callback for switching publishable
+        pyblish.register_callback(
+            "instanceToggled", on_pyblish_instance_toggled)
+
+        # install menu
+        menu.menu_install()
+        menu.add_scripts_menu()
+
+        # register hiero events
+        events.register_hiero_events()
+
+    def uninstall(self):
+        """
+        Uninstalling Hiero integration for avalon
+
+        """
+        log.info("Deregistering Hiero plug-ins..")
+        pyblish.deregister_host("hiero")
+        pyblish.deregister_plugin_path(PUBLISH_PATH)
+        deregister_loader_plugin_path(LOAD_PATH)
+        deregister_creator_plugin_path(CREATE_PATH)
+
+        # register callback for switching publishable
+        pyblish.deregister_callback("instanceToggled", on_pyblish_instance_toggled)
+
+    def get_context_data(self):
+        # root_node = nuke.root()
+        # return get_node_data(root_node, ROOT_DATA_KNOB)
+        pass
 
-    # register callback for switching publishable
-    pyblish.deregister_callback("instanceToggled", on_pyblish_instance_toggled)
+    def update_context_data(self, data, changes):
+        # root_node = nuke.root()
+        # set_node_data(root_node, ROOT_DATA_KNOB, data)
+        pass
```

Comment:
TODO - needs implementing otherwise Context attributes in publisher UI will not persist.